syntax = "proto3";

package embeddings;

option go_package = "./embeddingspb";

service Embeddings {
  rpc GenerateEmbeddings (EmbeddingsRequest) returns (EmbeddingsResponse);
}

message EmbeddingsRequest {
  repeated int64 token_ids = 1;
  string text = 2;
  string uuid = 3;
}

message EmbeddingsResponse {
  bool success = 1;
  string error_message = 2; // Optional error message if success is false
}
